### PR TITLE
Support Aggregations 2.0 Asset, Events, Time Series, Sequences

### PR DESCRIPTION
**Pull request by doctrino**
## Description
**Merge After** #1308 and #1303 

### Added
- Extended aggregation support for `Events`, `TimeSeries`, `Assets` and `Sequences`. This is available through the 
  methods `.aggregate_count(...)`, `aggregate_cardinality(...)`, and `.aggregate_unique(...)`. For example, 
  `client.events.aggregate_count(...)`.

### Improved
- Added `DeprecationWarning` to methods `client.assets.aggregate_metadata_keys`, 
  `client.assets.aggregate_metadata_values`, and `client.events.aggregate_unique_values` as the endpoint
  used by these methods have been deprecated in the CDF API.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api/assets.py [hunk 1]
```diff
@@ -427,8 +432,188 @@ def aggregate_metadata_values(
...
+        else:
+            return self._aggregate2(
+                "cardinalityValues",
+                properties=property,
+                filter=filter,
+                advanced_filter=advanced_filter,
+                aggregate_filter=aggregate_filter,
+            )
+
+    def aggregate_unique(
```
Comment (erlendvollset) [inline cognite/client/_api/assets.py:542]: These need to be split into `aggregate_unique_values` and `aggregate_unique_properties`. Same for cardinality aggregates. Should be able to aggregate values for metadata fields.

File: tests/tests_integration/test_api/test_assets.py [hunk 2]
```diff
@@ -307,6 +307,48 @@ def test_filter_on_metadata_key(self, cognite_client: CogniteClient, asset_list:
...
+
+        # Assert
+        assert count > 0, "Expected at one metadata key to exists"
+
+    def test_aggregate_unique_timezone(self, cognite_client: CogniteClient, asset_list: AssetList) -> None:
+        # Act
+        result = cognite_client.assets.aggregate_unique(AssetProperty.metadata_key("timezone"))
+
+        # Assert
+        assert len(result.unique) > 0, "Expected a least one timezone to exists"
```
Comment (erlendvollset) [inline tests/tests_integration/test_api/test_assets.py:343]: Let's be more thorough than asserting only length of result for these.

File: cognite/client/_api/sequences.py [hunk 3]
```diff
@@ -287,6 +288,198 @@ def aggregate(self, filter: Optional[Union[SequenceFilter, Dict]] = None) -> Lis
...
+            >>> from cognite.client.data_classes import filters, aggregations
+            >>> from cognite.client.data_classes.sequences import SequenceProperty
+            >>> c = CogniteClient()
+            >>> a = aggregations
+            >>> not_america = a.Not(a.Prefix("america"))
+            >>> is_critical = filters.Search(SequenceProperty.description, "critical")
+            >>> timezone_count = c.sequences.aggregate_cardinality(
+            ...                                           SequenceProperty.metadata_key("timezone"),
+            ...                                           advanced_filter=is_critical,
+            ...                                           aggregate_filter=not_america)
```
Comment (erlendvollset) [inline cognite/client/_api/sequences.py:376]: That's a lot of whitespace 😁

File: cognite/client/_api/time_series.py [hunk 4]
```diff
@@ -320,6 +321,198 @@ def aggregate(self, filter: Optional[Union[TimeSeriesFilter, Dict]] = None) -> L
...
+            >>> from cognite.client.data_classes import filters, aggregations
+            >>> from cognite.client.data_classes.time_series import TimeSeriesProperty
+            >>> c = CogniteClient()
+            >>> a = aggregations
+            >>> not_america = a.Not(a.Prefix("america"))
+            >>> is_critical = filters.Search(TimeSeriesProperty.description, "critical")
+            >>> timezone_count = c.time_series.aggregate_cardinality(
+            ...                                           TimeSeriesProperty.metadata_key("timezone"),
+            ...                                           advanced_filter=is_critical,
+            ...                                           aggregate_filter=not_america)
```
Comment (erlendvollset) [inline cognite/client/_api/time_series.py:409]: maybe chill with the space here too 😁 👾

File: cognite/client/data_classes/aggregations.py [hunk 5]
```diff
@@ -261,3 +282,66 @@ def _filter_body(self) -> dict:
...
+            body["gte"] = self._gte
+        if self._lt is not None:
+            body["lt"] = self._lt
+        if self._lte is not None:
+            body["lte"] = self._lte
+        return body
+
+
+@dataclass
+class UniqueResult(CogniteResource):
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/client/data_classes/aggregations.py:322]: The thinking was to be able to add helper methods to `UniqueResultList` for example `unique` which I demonstrate in the tests.

**Pull request merged**